src/main/java/org/reactome/release/common/dataretrieval/FileRetriever.java

```diff
@@ -0,0 +1,364 @@
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.SocketException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.time.Duration;
+import java.time.Instant;
+
+import org.apache.commons.net.ftp.FTP;
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.apache.http.conn.HttpHostConnectException;
+import org.apache.http.conn.UnsupportedSchemeException;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
+
+public class FileRetriever implements DataRetriever {
+
+	protected URI uri;
+	protected String destination;
+	protected Duration maxAge;
+	protected Duration timeout = Duration.ofSeconds(30);
+	protected int numRetries = 1;
+	protected String retrieverName;
+	protected Logger logger;
+	protected boolean passiveFTP = false;
+	
+	
+	public FileRetriever()
+	{
+		this(null);
+	}
+	
+	/**
+	 * This constructor takes a string that will be used to name the logging output file.
+	 * 
+	 * @param retrieverName - the name of this File Retriever - the log file produced by this class will be named "${retrieverName}.log"
+	 */
+	public FileRetriever(String retrieverName)
+	{
+		//logger.trace("Setting retrieverName to {}", retrieverName);
+		this.setRetrieverName(retrieverName);
+		this.logger = this.createLogger(this.retrieverName, "RollingRandomAccessFile", this.getClass().getName(), true, Level.DEBUG, this.logger, "Data Retriever");
+	}
+	
+	@Override
+	public void fetchData() throws Exception 
+	{
+		if (this.uri == null)
+		{
+			throw new RuntimeException("You must provide a URI from which the file will be downloaded!");
+		}
+		else if (this.destination.trim().length() == 0)
+		{
+			throw new RuntimeException("You must provide a destination to which the file will be downloaded!");
+		}
+		//Before fetching anything, we need to check to see if the file already exists.
+		Path pathToFile = Paths.get(this.destination);
+		if (Files.exists(pathToFile))
+		{
+			BasicFileAttributes attributes = Files.readAttributes(pathToFile, BasicFileAttributes.class);
+			
+			Instant fileCreateTime = attributes.creationTime().toInstant();
+			Instant now = Instant.now();
+			//If the file is older than the maxAge...
+			if (fileCreateTime.isBefore( now.minus(this.maxAge) ))
```

Comment:
Will maxAge be configurable or is there a hard-coded value somewhere?

Reply:
Yes, there is a `setMaxAge` method.